spark/src/main/java/com/robinhood/spark/SparkPaths.java

```diff
@@ -0,0 +1,59 @@
+package com.robinhood.spark;
+
+import android.support.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SparkPaths {
```

Comment:
Given the complexity of `SparkPaths` and `SparkPath`, I feel guilty about not unit testing it. The reason I know it's working is because of extensive manual testing (because of all the `SparkView` and animations-related behaviour that can't really be unit tested). Given even the tiniest push, I'll probably write unit tests for these two classes, but I expect `SparkPathSegment` to be tricky because it extends `andrioid.graphics.Path`.

Reply:
As an addendum: I wanted to write most of this in Kotlin, or at least make this library use Java 8, but figured if possible we should try to keep backwards compatibility (and neither Kotlin nor Java 8 are supported by this library in its current configuration). If we don't care _at all_ and just plan on using it internally, one or both of these can be done in a subsequent PR.